visivo/parsers/mkdocs_utils/nav_configuration_generator.py

```diff
@@ -0,0 +1,235 @@
+
+
+def _to_mkdocs_yaml(schema, structure, base_path="reference/configuration"):
+    output = []
+
+    for model, contents in structure.items():
+        if isinstance(contents, dict):
+            sub_path = f"{base_path}/{model}"
+            if schema.get("$defs", {}).get(model, {}):
+                model_content = {
+                    model: [f"{sub_path}/index.md"]
+                    + _to_mkdocs_yaml(schema, contents, sub_path)
+                }
+            else:
+                model_content = {model: _to_mkdocs_yaml(schema, contents, sub_path)}
+            output.append(model_content)
+        # Handle other cases if necessary (e.g., primitive types)
+
+    return output
+
+
+def _pop_nested_path(dictionary: dict, path: list):
+    """Modifies dictionary removing key from a nested path, ignoring nested paths that may have already been removed"""
+    result = dictionary
+    for key in path[:-1]:
+        if isinstance(result, dict) and key in result.keys():
+            result = result[key]
+
+    if path[-1] in result.keys():
+        result.pop(path[-1])
+
+
+def _pop_list_of_nested_paths(dictionary: dict, paths: list):
+    for path in paths:
+        _pop_nested_path(dictionary, path)
+
+
+def _get_all_key_paths(dictionary, path=None) -> list:
+    items = []
+    path = path or []
+    for key, value in dictionary.items():
+        new_path = path + [key]
+        if isinstance(value, dict):
+            items.append({key: new_path})
+            items += _get_all_key_paths(value, new_path)
+
+    return items
+
+
+def _consolidate_paths(paths: list) -> dict:
+    consolidated = {}
+    for dictionary in paths:
+        model = list(dictionary.keys())[0]
+        path = list(dictionary.values())[0]
+        if model not in consolidated.keys():
+            consolidated[model] = [path]
+        elif model in consolidated.keys():
+            consolidated[model].append(path)
+    return consolidated
+
+
+def _get_paths_to_remove(consolidated_paths: dict) -> list:
+    to_remove = []
+    for model, paths in consolidated_paths.items():
+
+        top_path_length = 9999
+        # find the shortest path to model
+        for path in paths:
+            if len(path) < top_path_length:
+                top_path_length = len(path)
+        # add longer paths to to_remove_list
+        for path in paths:
+            if len(path) != top_path_length:
+                to_remove.append(path)
+    return to_remove
+
+
+def mkdocs_pydantic_nav(schema: dict) -> list:
+    nested_structure = _generate_structure(schema)
+    all_key_paths = _get_all_key_paths(nested_structure)
+    consolidated_paths = _consolidate_paths(all_key_paths)
+    paths_to_remove = _get_paths_to_remove(consolidated_paths)
+    _pop_list_of_nested_paths(nested_structure, paths=paths_to_remove)
+    yaml_output = _to_mkdocs_yaml(schema, nested_structure)
+    return yaml_output
+
+
+def _extract_strings_from_yaml(yaml_obj):
```

Comment:
I guess they have to be first in python.  Does make it difficult to know the where to start in a file like this one.

Reply:
Ya, it took a lot of recursion to navigate the pydantic & mkdocs objects because of the lists.